From fb47acb592e7e201e5ccedbfbae7d32d8cf254f2 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Wed, 9 Mar 2011 15:52:26 -0500
Subject: [PATCH] daemon: fix crash with autologin

We were using the wrong function prototype which was leading
to crashes.
---
 daemon/gdm-session-worker.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/daemon/gdm-session-worker.c b/daemon/gdm-session-worker.c
index 299c8be..15ec0e9 100644
--- a/daemon/gdm-session-worker.c
+++ b/daemon/gdm-session-worker.c
@@ -2094,7 +2094,9 @@ save_account_details_now (GdmSessionWorker *worker)
 }
 
 static void
-on_settings_is_loaded_changed (GdmSessionWorker *worker)
+on_settings_is_loaded_changed (GdmSessionSettings *user_settings,
+                               GParamSpec         *pspec,
+                               GdmSessionWorker   *worker)
 {
         if (!gdm_session_settings_is_loaded (worker->priv->user_settings)) {
                 return;
-- 
1.7.4.1

